source/SH3/arc/vfile.cpp

```diff
@@ -100,3 +102,23 @@ std::size_t sh3_arc_vfile::ReadData(void* destination, std::size_t len, read_err
 
     return nbytes;
 }
+
+void sh3_arc_vfile::Dump2Disk()
+{
+    if(!open || buffer.empty())
+    {
+        Log(LogLevel::WARN, "sh3_arc_vfile::Dump2Disk( ): Warning! Attempting to flush unopen or empty buffer to disk!");
+        return;
+    }
+
+    // Firstly, we need to convert the file path to just the filename.
+    std::stringstream ss(fname);
+    std::string item;
+
+    while(getline(ss, item, '/')); // Throw out all strings except the last one (our file name)
+
+    std::ofstream out_file(item);
+    for(int i = 0; i < buffer.size(); i++)
+        out_file << buffer[i];
```

Comment:
The "streaming operators" (`<<`, `>>`) are meant for formatted data. We're dealing with binary data, so use `std::ofstream::write` instead. And you don't need to do it item by item then, you can just `write(buffer.data(), buffer.size())`.